Lib/test/test_msvcrt.py

```diff
@@ -61,34 +61,48 @@ def test_get_osfhandle(self):
 
 
 class TestConsoleIO(unittest.TestCase):
+    def run_in_separated_process(self, code):
+        # Run test in a seprated process to avoid stdin conflicts.
+        # See: gh-110147
+        cmd = [sys.executable, '-c', code]
+        subprocess.run(cmd, check=True, capture_output=True,
+                       creationflags=subprocess.CREATE_NEW_CONSOLE)
+
+
     def test_kbhit(self):
-        h = msvcrt.get_osfhandle(sys.stdin.fileno())
-        flush_console_input_buffer(h)
-        self.assertEqual(msvcrt.kbhit(), 0)
+        code = dedent('''
+            import msvcrt
+            assert msvcrt.kbhit() == 0
+        ''')
+        self.run_in_separated_process(code)
 
     def test_getch(self):
         msvcrt.ungetch(b'c')
         self.assertEqual(msvcrt.getch(), b'c')
 
     def test_getwch(self):
-        with open('CONIN$', 'rb', buffering=0) as stdin:
-            h = msvcrt.get_osfhandle(stdin.fileno())
-            flush_console_input_buffer(h)
-
-            write_input(stdin, c_encoded)
-            self.assertEqual(msvcrt.getwch(), c)
+        code = dedent(f'''
+            import msvcrt
+            from _testconsole import write_input
+            with open("CONIN$", "rb", buffering=0) as stdin:
+                write_input(stdin, {repr(c_encoded)})
```

Comment:
```suggestion
                write_input(stdin, {ascii(c_encoded)})
```

Same remark for test_getwche().